worker-message-prioritization-target-capacity-calculators/src/main/java/com/github/workerframework/workermessageprioritization/targetqueue/K8sTargetQueueSettingsProvider.java

```diff
@@ -48,29 +50,20 @@ public final class K8sTargetQueueSettingsProvider implements TargetQueueSettings
     private static final String MESSAGE_PRIORITIZATION_TARGET_QUEUE_ELIGIBLE_FOR_REFILL_PERCENTAGE_LABEL
         = "messageprioritization.targetqueueeligibleforrefillpercentage";
     private static final String CURRENT_INSTANCES_LABEL = "spec.replicas";
-    private static final long TARGET_QUEUE_MAX_LENGTH_FALLBACK = 1000;
-    private static final long TARGET_QUEUE_ELIGIBLE_FOR_REFILL_PERCENTAGE_FALLBACK = 10;
-    private static final int CURRENT_INSTANCE_FALLBACK = 1;
-    private static final int MAX_INSTANCES_FALLBACK = 1;
-    private static final long CAPACITY_FALLBACK = 1000;
-    private static final TargetQueueSettings FALLBACK_TARGET_QUEUE_SETTINGS = new TargetQueueSettings(
-            TARGET_QUEUE_MAX_LENGTH_FALLBACK,
-            TARGET_QUEUE_ELIGIBLE_FOR_REFILL_PERCENTAGE_FALLBACK,
-            MAX_INSTANCES_FALLBACK,
-            CURRENT_INSTANCE_FALLBACK,
-            CAPACITY_FALLBACK);
     private final List<String> kubernetesNamespaces;
     private final Supplier<Map<String, TargetQueueSettings>> memoizedTargetQueueSettingsSupplier;
+    private final AppsV1Api appsV1Api;
 
     @Inject
     public K8sTargetQueueSettingsProvider(
             @Named("KubernetesNamespaces") final List<String> kubernetesNamespaces, 
             @Named("KubernetesLabelCacheExpiryMinutes") final int kubernetesLabelCacheExpiryMinutes)
     {
         try {
-            Configuration.setDefaultApiClient(ClientBuilder.standard().build());
-        } catch (final IOException ioException) {
-            throw new RuntimeException("IOException thrown trying to create a Kubernetes client", ioException);
+            final ApiClient clientWithCertAndToken = KubernetesClientFactory.createClientWithCertAndToken();
+            this.appsV1Api = new AppsV1Api(clientWithCertAndToken);
+        } catch (final Exception e) {
```

Comment:
I haven't looked at the new client code yet to understand this but the need to move to catching top-level exception might be indicating an issue.

Reply:
The old client did not throw an exception if the following env vars were not set:

```
KUBERNETES_SERVICE_HOST
KUBERNETES_SERVICE_PORT
```

In the new client I throw a RuntimeException if these are not set, as the new client will not work without them:

https://github.com/CAFapi/caf-kubernetes-client/pull/1/files#diff-5e074d6be6662cb7cc86dc4fa322eb18456b58226ab2ceed27b243d40361f729R79

Note the old client also did not work without them, but it failed at runtime rather than construction time. If you look at the integration test logs you can see the calls to k8s failing at runtime (we catch the exception and then use fallback settings):

https://sou-jenkins2.swinfra.net/job/WorkerFramework/job/WorkerFramework%7Eworker-message-prioritization%7Emain%7ECI/161/consoleFull

```
2024-08-25T04:21:19.018656889Z worker-message-distribution> [2024-08-25 04:21:19.013Z #7f9.001 ERROR -            -   ] c.g.w.w.t.K8sTargetQueueSettingsProvider:[  ]{"message":"Cannot get settings for the target queues in the private namespace as the Kubernetes API threw an exception.","exception":"java.net.ConnectException: Connection refused
```

I thought it was cleaner to do it this way, e.g. fail fast at construction time rather than fail at runtime?